File: Leetcode/Java/Easy/3622CheckDivisibilitybyDigitSumandProduct.java
class Solution{
    public boolean checkDivisibility(int n){
        if(n == 0){
            return false;
        }
        int product = 1;
        int sum = 0;
        String string_n = Integer.toString(n);
        for(int i =0; i < string_n.length(); i++){
            
            sum += Integer.valueOf(string_n.charAt(i) - '0');
            product *= Integer.valueOf(string_n.charAt(i) - '0');
        }
        int total_sum = sum + product;
        if(n % total_sum == 0){
            return true;
        }else{
            return false;
        }
    }
}
//Accepted solution

File: Leetcode/PythonAnswers/Easy/3622CheckDivisibilitybyDigitSumandProduct.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        if n == 0:
            return False
        sum = 0
        product = 1
        string_n = str(n)
        for i in string_n:
            sum += int(i)
            product *= int(i)
        total = product + sum
        if(n % total == 0):
            return True
        else:
            return False
#Accepted solution